### ISSUE TITLE
[BUG]: Streaming `logprob` & `echo` combo.

### ISSUE THREAD

**Issue by AguirreNicolas** [vllm-project/vllm#2703]
I'm trying to start writing a `logprob` & `echo` support for chat request. 

Unfortunately, running test like #1992 when `echo` is setted as `true` server doesn't respond.

Seeing furtherer I checked that the **bug** begging in #2449 (sha: dd7e8f5f643167e3f13045cf75cbead54cb2ccfe).
Previous commit #2463 (sha: d2a68364c473a3167a1c2b90f947bb611322a867) worked ok.

## LOG:
```
vllm-openai-main  | INFO 02-01 04:31:38 async_llm_engine.py:385] Received request cmpl-dc7fb40d1b534a879768966f3dc50d39: prompt: None, prefix_pos: None,sampling params: SamplingParams(n=1, best_of=1, presence_penalty=0.0, frequency_penalty=0.0, repetition_penalty=1.0, temperature=1.0, top_p=1.0, top_k=-1, min_p=0.0, use_beam_search=False, length_penalty=1.0, early_stopping=False, stop=[], stop_token_ids=[], include_stop_str_in_output=False, ignore_eos=True, max_tokens=20, logprobs=0, prompt_logprobs=0, skip_special_tokens=True, spaces_between_special_tokens=True), prompt token ids: [2, 12375, 351, 5, 232, 651, 11, 2760, 116, 50118, 6557, 45117, 35, 50118].
vllm-openai-main  | ERROR:    Exception in ASGI application
vllm-openai-main  | Traceback (most recent call last):
vllm-openai-main  |   File "/usr/local/lib/python3.10/dist-packages/starlette/responses.py", line 259, in __call__
vllm-openai-main  |     await wrap(partial(self.listen_for_disconnect, receive))
vllm-openai-main  |   File "/usr/local/lib/python3.10/dist-packages/starlette/responses.py", line 255, in wrap
vllm-openai-main  |     await func()
vllm-openai-main  |   File "/usr/local/lib/python3.10/dist-packages/starlette/responses.py", line 232, in listen_for_disconnect
vllm-openai-main  |     message = await receive()
vllm-openai-main  |   File "/usr/local/lib/python3.10/dist-packages/uvicorn/protocols/http/httptools_impl.py", line 587, in receive
vllm-openai-main  |     await self.message_event.wait()
vllm-openai-main  |   File "/usr/lib/python3.10/asyncio/locks.py", line 214, in wait
vllm-openai-main  |     await fut
vllm-openai-main  | asyncio.exceptions.CancelledError: Cancelled by cancel scope 7f20005d5150
vllm-openai-main  | 
vllm-openai-main  | During handling of the above exception, another exception occurred:
vllm-openai-main  | 
vllm-openai-main  | Traceback (most recent call last):
vllm-openai-main  |   File "/usr/local/lib/python3.10/dist-packages/uvicorn/protocols/http/httptools_impl.py", line 426, in run_asgi
vllm-openai-main  |     result = await app(  # type: ignore[func-returns-value]
vllm-openai-main  |   File "/usr/local/lib/python3.10/dist-packages/uvicorn/middleware/proxy_headers.py", line 84, in __call__
vllm-openai-main  |     return await self.app(scope, receive, send)
vllm-openai-main  |   File "/usr/local/lib/python3.10/dist-packages/fastapi/applications.py", line 1054, in __call__
vllm-openai-main  |     await super().__call__(scope, receive, send)
vllm-openai-main  |   File "/usr/local/lib/python3.10/dist-packages/starlette/applications.py", line 116, in __call__
vllm-openai-main  |     await self.middleware_stack(scope, receive, send)
vllm-openai-main  |   File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/errors.py", line 186, in __call__
vllm-openai-main  |     raise exc
vllm-openai-main  |   File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/errors.py", line 164, in __call__
vllm-openai-main  |     await self.app(scope, receive, _send)
vllm-openai-main  |   File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/cors.py", line 83, in __call__
vllm-openai-main  |     await self.app(scope, receive, send)
vllm-openai-main  |   File "/usr/local/lib/python3.10/dist-packages/aioprometheus/asgi/middleware.py", line 184, in __call__
vllm-openai-main  |     await self.asgi_callable(scope, receive, wrapped_send)
vllm-openai-main  |   File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/exceptions.py", line 62, in __call__
vllm-openai-main  |     await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
vllm-openai-main  |   File "/usr/local/lib/python3.10/dist-packages/starlette/_exception_handler.py", line 55, in wrapped_app
vllm-openai-main  |     raise exc
vllm-openai-main  |   File "/usr/local/lib/python3.10/dist-packages/starlette/_exception_handler.py", line 44, in wrapped_app
vllm-openai-main  |     await app(scope, receive, sender)
vllm-openai-main  |   File "/usr/local/lib/python3.10/dist-packages/starlette/routing.py", line 746, in __call__
vllm-openai-main  |     await route.handle(scope, receive, send)
vllm-openai-main  |   File "/usr/local/lib/python3.10/dist-packages/starlette/routing.py", line 288, in handle
vllm-openai-main  |     await self.app(scope, receive, send)
vllm-openai-main  |   File "/usr/local/lib/python3.10/dist-packages/starlette/routing.py", line 75, in app
vllm-openai-main  |     await wrap_app_handling_exceptions(app, request)(scope, receive, send)
vllm-openai-main  |   File "/usr/local/lib/python3.10/dist-packages/starlette/_exception_handler.py", line 55, in wrapped_app
vllm-openai-main  |     raise exc
vllm-openai-main  |   File "/usr/local/lib/python3.10/dist-packages/starlette/_exception_handler.py", line 44, in wrapped_app
vllm-openai-main  |     await app(scope, receive, sender)
vllm-openai-main  |   File "/usr/local/lib/python3.10/dist-packages/starlette/routing.py", line 73, in app
vllm-openai-main  |     await response(scope, receive, send)
vllm-openai-main  |   File "/usr/local/lib/python3.10/dist-packages/starlette/responses.py", line 252, in __call__
vllm-openai-main  |     async with anyio.create_task_group() as task_group:
vllm-openai-main  |   File "/usr/local/lib/python3.10/dist-packages/anyio/_backends/_asyncio.py", line 678, in __aexit__
vllm-openai-main  |     raise BaseExceptionGroup(
vllm-openai-main  | exceptiongroup.ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
vllm-openai-main  | INFO 02-01 04:31:38 async_llm_engine.py:111] Finished request cmpl-dc7fb40d1b534a879768966f3dc50d39.
```

**Comment by yuanphoenix** [comment]
same question

**Comment by piojanu** [comment]
Hi!

TL;DR: There is a regression somewhere between v0.4.2 and v0.6.6.post1 (current latest).

I experience this bug again in the v0.6.6.post1 (the latest at the moment). When I run the `Phi-3-mini-4k-instruct` model with this command `python3 -m vllm.entrypoints.openai.api_server --model microsoft/Phi-3-mini-4k-instruct --tensor-parallel-size 1` on the linux server, then this request `curl -N -X POST --url http://0.0.0.0:8000/v1/completions -H 'Content-Type: application/json' -d '{"prompt": "Warsaw is", "top_p": 1.0, "max_tokens": 20, "stream": false, "model": "microsoft/Phi-3-mini-4k-instruct", "temperature": 1.0, "logprobs": 1, "echo": true, "stream": true}'`  gives an error.

Response: `curl: (18) transfer closed with outstanding read data remaining`

Server-side error:

```
INFO 01-16 06:15:44 logger.py:37] Received request cmpl-3d6aff98ffad4a539ffadc2fb604fb10-0: prompt: 'Warsaw is', params: SamplingParams(n=1, presence_penalty=0.0, frequency_penalty=0.0, repetition_penalty=1.0, temperature=1.0, top_p=1.0, top_k=-1, min_p=0.0, seed=None, stop=[], stop_token_ids=[], bad_words=[], include_stop_str_in_output=False, ignore_eos=False, max_tokens=20, min_tokens=0, logprobs=1, prompt_logprobs=1, skip_special_tokens=True, spaces_between_special_tokens=True, truncate_prompt_tokens=None, guided_decoding=None), prompt_token_ids: [9149, 1450, 338], lora_request: None, prompt_adapter_request: None.
INFO:     10.52.60.13:40302 - "POST /v1/completions HTTP/1.1" 200 OK
INFO 01-16 06:15:44 engine.py:267] Added request cmpl-3d6aff98ffad4a539ffadc2fb604fb10-0.
INFO 01-16 06:15:44 metrics.py:467] Avg prompt throughput: 0.4 tokens/s, Avg generation throughput: 0.1 tokens/s, Running: 1 reqs, Swapped: 0 reqs, Pending: 0 reqs, GPU KV cache usage: 0.0%, CPU KV cache usage: 0.0%.
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/dist-packages/starlette/responses.py", line 259, in __call__
    await wrap(partial(self.listen_for_disconnect, receive))
  File "/usr/local/lib/python3.12/dist-packages/starlette/responses.py", line 255, in wrap
    await func()
  File "/usr/local/lib/python3.12/dist-packages/starlette/responses.py", line 232, in listen_for_disconnect
    message = await receive()
              ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/dist-packages/uvicorn/protocols/http/httptools_impl.py", line 563, in receive
    await self.message_event.wait()
  File "/usr/lib/python3.12/asyncio/locks.py", line 212, in wait
    await fut
asyncio.exceptions.CancelledError: Cancelled by cancel scope 7ffe706f84d0

During handling of the above exception, another exception occurred:

  + Exception Group Traceback (most recent call last):
  |   File "/usr/local/lib/python3.12/dist-packages/uvicorn/protocols/http/httptools_impl.py", line 409, in run_asgi
  |     result = await app(  # type: ignore[func-returns-value]
  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |   File "/usr/local/lib/python3.12/dist-packages/uvicorn/middleware/proxy_headers.py", line 60, in __call__
  |     return await self.app(scope, receive, send)
  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |   File "/usr/local/lib/python3.12/dist-packages/fastapi/applications.py", line 1054, in __call__
  |     await super().__call__(scope, receive, send)
  |   File "/usr/local/lib/python3.12/dist-packages/starlette/applications.py", line 113, in __call__
  |     await self.middleware_stack(scope, receive, send)
  |   File "/usr/local/lib/python3.12/dist-packages/starlette/middleware/errors.py", line 187, in __call__
  |     raise exc
  |   File "/usr/local/lib/python3.12/dist-packages/starlette/middleware/errors.py", line 165, in __call__
  |     await self.app(scope, receive, _send)
  |   File "/usr/local/lib/python3.12/dist-packages/starlette/middleware/cors.py", line 85, in __call__
  |     await self.app(scope, receive, send)
  |   File "/usr/local/lib/python3.12/dist-packages/starlette/middleware/exceptions.py", line 62, in __call__
  |     await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  |   File "/usr/local/lib/python3.12/dist-packages/starlette/_exception_handler.py", line 53, in wrapped_app
  |     raise exc
  |   File "/usr/local/lib/python3.12/dist-packages/starlette/_exception_handler.py", line 42, in wrapped_app
  |     await app(scope, receive, sender)
  |   File "/usr/local/lib/python3.12/dist-packages/starlette/routing.py", line 715, in __call__
  |     await self.middleware_stack(scope, receive, send)
  |   File "/usr/local/lib/python3.12/dist-packages/starlette/routing.py", line 735, in app
  |     await route.handle(scope, receive, send)
  |   File "/usr/local/lib/python3.12/dist-packages/starlette/routing.py", line 288, in handle
  |     await self.app(scope, receive, send)
  |   File "/usr/local/lib/python3.12/dist-packages/starlette/routing.py", line 76, in app
  |     await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  |   File "/usr/local/lib/python3.12/dist-packages/starlette/_exception_handler.py", line 53, in wrapped_app
  |     raise exc
  |   File "/usr/local/lib/python3.12/dist-packages/starlette/_exception_handler.py", line 42, in wrapped_app
  |     await app(scope, receive, sender)
  |   File "/usr/local/lib/python3.12/dist-packages/starlette/routing.py", line 74, in app
  |     await response(scope, receive, send)
  |   File "/usr/local/lib/python3.12/dist-packages/starlette/responses.py", line 252, in __call__
  |     async with anyio.create_task_group() as task_group:
  |                ^^^^^^^^^^^^^^^^^^^^^^^^^
  |   File "/usr/local/lib/python3.12/dist-packages/anyio/_backends/_asyncio.py", line 815, in __aexit__
  |     raise BaseExceptionGroup(
  | ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
  +-+---------------- 1 ----------------
    | Traceback (most recent call last):
    |   File "/usr/local/lib/python3.12/dist-packages/starlette/responses.py", line 255, in wrap
    |     await func()
    |   File "/usr/local/lib/python3.12/dist-packages/starlette/responses.py", line 244, in stream_response
    |     async for chunk in self.body_iterator:
    |   File "/usr/local/lib/python3.12/dist-packages/vllm/entrypoints/openai/serving_completion.py", line 286, in completion_stream_generator
    |     assert prompt_text is not None
    |            ^^^^^^^^^^^^^^^^^^^^^^^
    | AssertionError
    +------------------------------------
INFO 01-16 06:15:44 engine.py:285] Aborted request cmpl-3d6aff98ffad4a539ffadc2fb604fb10-0.
INFO 01-16 06:15:54 metrics.py:467] Avg prompt throughput: 0.0 tokens/s, Avg generation throughput: 0.0 tokens/s, Running: 0 reqs, Swapped: 0 reqs, Pending: 0 reqs, GPU KV cache usage: 0.0%, CPU KV cache usage: 0.0%.
```

I found the first build that has both this fix and supports Phi-3-mini model, which is v0.4.2, and the same request returns proper response:

```
data: {"id":"cmpl-9bdca2835ba94646878d6ca04c9d7595","created":1737036781,"model":"microsoft/Phi-3-mini-4k-instruct","choices":[{"index":0,"text":"Warsaw is at","logprobs":{"top_logprobs":[null,{"aw":-2.4762144088745117,",":-1.9762145280838013},{"is":-4.902688503265381,",":-1.6526883840560913},{" at":-5.260345458984375," the":-1.760345697402954}]},"finish_reason":null,"stop_reason":null}],"usage":null}

data: {"id":"cmpl-9bdca2835ba94646878d6ca04c9d7595","created":1737036781,"model":"microsoft/Phi-3-mini-4k-instruct","choices":[{"index":0,"text":" coordinates","logprobs":{"top_logprobs":[{" coordinates":-3.5154056549072266," the":-1.015405535697937}]},"finish_reason":null,"stop_reason":null}],"usage":null}

data: {"id":"cmpl-9bdca2835ba94646878d6ca04c9d7595","created":1737036781,"model":"microsoft/Phi-3-mini-4k-instruct","choices":[{"index":0,"text":" [","logprobs":{"top_logprobs":[{" [":-5.4179253578186035," (":-0.16792547702789307}]},"finish_reason":null,"stop_reason":null}],"usage":null}

data: {"id":"cmpl-9bdca2835ba94646878d6ca04c9d7595","created":1737036781,"model":"microsoft/Phi-3-mini-4k-instruct","choices":[{"index":0,"text":"3","logprobs":{"top_logprobs":[{"3":-3.4139633178710938,"5":-0.913963258266449}]},"finish_reason":null,"stop_reason":null}],"usage":null}

data: {"id":"cmpl-9bdca2835ba94646878d6ca04c9d7595","created":1737036781,"model":"microsoft/Phi-3-mini-4k-instruct","choices":[{"index":0,"text":"8","logprobs":{"top_logprobs":[{"8":-2.295501708984375,"0":-1.5455015897750854}]},"finish_reason":null,"stop_reason":null}],"usage":null}

data: {"id":"cmpl-9bdca2835ba94646878d6ca04c9d7595","created":1737036781,"model":"microsoft/Phi-3-mini-4k-instruct","choices":[{"index":0,"text":".","logprobs":{"top_logprobs":[{".":-0.24965709447860718}]},"finish_reason":null,"stop_reason":null}],"usage":null}

data: {"id":"cmpl-9bdca2835ba94646878d6ca04c9d7595","created":1737036781,"model":"microsoft/Phi-3-mini-4k-instruct","choices":[{"index":0,"text":"7","logprobs":{"top_logprobs":[{"7":-0.8204102516174316}]},"finish_reason":null,"stop_reason":null}],"usage":null}

data: {"id":"cmpl-9bdca2835ba94646878d6ca04c9d7595","created":1737036781,"model":"microsoft/Phi-3-mini-4k-instruct","choices":[{"index":0,"text":"0","logprobs":{"top_logprobs":[{"0":-2.6188926696777344,"2":-0.6188927888870239}]},"finish_reason":null,"stop_reason":null}],"usage":null}

data: {"id":"cmpl-9bdca2835ba94646878d6ca04c9d7595","created":1737036781,"model":"microsoft/Phi-3-mini-4k-instruct","choices":[{"index":0,"text":"6","logprobs":{"top_logprobs":[{"6":-2.7134783267974854,"4":-0.9634783267974854}]},"finish_reason":null,"stop_reason":null}],"usage":null}

data: {"id":"cmpl-9bdca2835ba94646878d6ca04c9d7595","created":1737036781,"model":"microsoft/Phi-3-mini-4k-instruct","choices":[{"index":0,"text":"0","logprobs":{"top_logprobs":[{"0":-2.281411647796631,"2":-1.5314117670059204}]},"finish_reason":null,"stop_reason":null}],"usage":null}

data: {"id":"cmpl-9bdca2835ba94646878d6ca04c9d7595","created":1737036781,"model":"microsoft/Phi-3-mini-4k-instruct","choices":[{"index":0,"text":",","logprobs":{"top_logprobs":[{",":-0.7764667272567749}]},"finish_reason":null,"stop_reason":null}],"usage":null}

data: {"id":"cmpl-9bdca2835ba94646878d6ca04c9d7595","created":1737036781,"model":"microsoft/Phi-3-mini-4k-instruct","choices":[{"index":0,"text":" ","logprobs":{"top_logprobs":[{" ":-0.13979218900203705}]},"finish_reason":null,"stop_reason":null}],"usage":null}

data: {"id":"cmpl-9bdca2835ba94646878d6ca04c9d7595","created":1737036781,"model":"microsoft/Phi-3-mini-4k-instruct","choices":[{"index":0,"text":"2","logprobs":{"top_logprobs":[{"2":-0.3091893494129181}]},"finish_reason":null,"stop_reason":null}],"usage":null}

data: {"id":"cmpl-9bdca2835ba94646878d6ca04c9d7595","created":1737036781,"model":"microsoft/Phi-3-mini-4k-instruct","choices":[{"index":0,"text":"1","logprobs":{"top_logprobs":[{"1":-0.21141889691352844}]},"finish_reason":null,"stop_reason":null}],"usage":null}

data: {"id":"cmpl-9bdca2835ba94646878d6ca04c9d7595","created":1737036781,"model":"microsoft/Phi-3-mini-4k-instruct","choices":[{"index":0,"text":".","logprobs":{"top_logprobs":[{".":-0.00039498155820183456}]},"finish_reason":null,"stop_reason":null}],"usage":null}

data: {"id":"cmpl-9bdca2835ba94646878d6ca04c9d7595","created":1737036781,"model":"microsoft/Phi-3-mini-4k-instruct","choices":[{"index":0,"text":"4","logprobs":{"top_logprobs":[{"4":-0.27460289001464844}]},"finish_reason":null,"stop_reason":null}],"usage":null}

data: {"id":"cmpl-9bdca2835ba94646878d6ca04c9d7595","created":1737036781,"model":"microsoft/Phi-3-mini-4k-instruct","choices":[{"index":0,"text":"4","logprobs":{"top_logprobs":[{"4":-1.1617751121520996}]},"finish_reason":null,"stop_reason":null}],"usage":null}

data: {"id":"cmpl-9bdca2835ba94646878d6ca04c9d7595","created":1737036781,"model":"microsoft/Phi-3-mini-4k-instruct","choices":[{"index":0,"text":"0","logprobs":{"top_logprobs":[{"0":-3.1873276233673096,"4":-0.9373276829719543}]},"finish_reason":null,"stop_reason":null}],"usage":null}

data: {"id":"cmpl-9bdca2835ba94646878d6ca04c9d7595","created":1737036781,"model":"microsoft/Phi-3-mini-4k-instruct","choices":[{"index":0,"text":"2","logprobs":{"top_logprobs":[{"2":-3.2756426334381104,"3":-0.7756425738334656}]},"finish_reason":null,"stop_reason":null}],"usage":null}

data: {"id":"cmpl-9bdca2835ba94646878d6ca04c9d7595","created":1737036781,"model":"microsoft/Phi-3-mini-4k-instruct","choices":[{"index":0,"text":"],","logprobs":{"top_logprobs":[{"],":-1.270097017288208,"].":-0.520097017288208}]},"finish_reason":"length","stop_reason":null}],"usage":{"prompt_tokens":3,"total_tokens":23,"completion_tokens":20}}

data: [DONE]
```

I can't reopen the issue, can you do this? Or should I create a new one?

Thanks!
